compiler/luci/import/src/CircleReader.cpp

```diff
@@ -349,6 +362,12 @@ bool CircleReader::select_subgraph(uint32_t sgindex)
   return true;
 }
 
+const uint8_t *CircleReader::file_data(uint64_t offset) const
+{
+  assert(_file_data);
```

Comment:
Null _file_data is handled below.

```suggestion
```

Reply:
BTW, is it possible to do range change? I think offset should be less than file size.

Reply:
this assert is necessary with calling existing 
`bool CircleReader::parse(const circle::Model *model)` method.
need to be sure.

Reply:
> to do range change?

Um... maybe range check? size can be used.
I think I can revise this.

Reply:
> this assert is necessary

I see. Then, it would be better to update the below code for consistent behavior of Release mode and Debug mode.

From
```cpp
  return (_file_data == nullptr) ? nullptr : _file_data + offset;
```

To
```cpp
  return _file_data + offset;
```

Reply:
checker is just assert for debug version.
release version caller will be killed with `return _file_data + offset;`

Reply:
>  consistent behavior of Release mode and Debug mode.

I don't understand this.

Reply:
I think that debug mode and release mode should behave in the same way. If you think that's ok, it would be fine to go as-is.

Reply:
Thanks for adding assert for range check.

Reply:
> I think that debug mode and release mode should behave in the same way.

I don't think we are implementing in the same way.
We're adding lots of `assert()` that gives different results.
Usage of `assert()` for me is condition check that doesn't(or shouldn't) happen in most(all) cases, as you may know. And to get early failure state in debug testing.
As `assert()` is nothing in release build, debug and release build behavior is different.

Or do you have different view?

Reply:
As you mentioned, `assert()` indicates a condition that should not happend in runtime. So I think if assert is violated, it's a bug.

The current function handles the case in release mode, which means that nullptr of `_file_data` is acceptable. So, bug will be silently passed in release mode. I guess that segfault may occur in some place where we don't know..

> release version caller will be killed with return _file_data + offset;

I think it is better to die in the place where problem exists. To avoid segfault, it would be possible to raise an exception and handle it.